packages/annotations/src/block/index.ts

```diff
@@ -0,0 +1,47 @@
+/**
+ * WordPress dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { withSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { STORE_NAME } from '../store/constants';
+import type { WPAnnotation } from '../types';
+
+/**
+ * Adds annotation className to the block-list-block component.
+ *
+ * @param OriginalComponent The original BlockListBlock component.
+ * @return The enhanced component.
+ */
+const addAnnotationClassName = ( OriginalComponent: any ) => {
+	return ( withSelect as any )(
```

Comment:
I would prefer to avoid these two `any` here, which means, we probably should first convert `withSelect` to typescript.